dbt/adapters/duckdb/relation.py

```diff
@@ -0,0 +1,25 @@
+from typing import Any, Optional, Type
+
+from dataclasses import dataclass
+
+from dbt.adapters.base.relation import BaseRelation, Self
+from dbt.contracts.graph.parsed import ParsedSourceDefinition
+
+
+@dataclass(frozen=True, eq=False, repr=False)
+class DuckDBRelation(BaseRelation):
+    external_location: Optional[str] = None
+
+    @classmethod
+    def create_from_source(cls: Type[Self], source: ParsedSourceDefinition, **kwargs: Any) -> Self:
+        if source.external:
+            # TODO(jwills): this is trivial and could be made much richer, just want to know
+            # if it works
+            kwargs["external_location"] = f"'{source.external.location}'"
+        return super().create_from_source(source, **kwargs)
```

Comment:
Ah yeah so this is my ignorance of the details of the impl of python classmethods-- how exactly do I signal that I want the call to go to the parent class' classmethod instead of this one? It's not, `super()`, it's `cls`? Really?

Reply:
Oh no, it is my bad I did not realize you are calling the same method name. I guess it should be `super(DuckDBRelation, cls)`, will take a look if mypy is ok with it.

Reply:
Oh does not help. But it seems like a known issue - https://github.com/python/mypy/issues/12885. So we can probably ignore:

```
return super().create_from_source(source, **kwargs)  # type: ignore
```